[Streams] Nested conditionals followups and simulation bug fix

Kerry350 opened the pull request:
## Summary

This predominantly deals with the followups in https://github.com/elastic/kibana/issues/235508. 

It also fixes a bug 🐞 with simulation — condition steps that were **already persisted** weren't being passed through for simulation, they are now.

Kerry350 committed:
Nested conditionals followups and simulation bug fix

elasticmachine commented:
Pinging @elastic/obs-ux-logs-team (Team:obs-ux-logs)

Kerry350 committed:
Merge remote-tracking branch 'upstream/main' into 235508-nested-conditionals-followups

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/generate codeowners'

tonyghiani reviewed (approved):
Changes LGTM, good catch on the condition bug! 👏

Kerry350 committed:
Amend i18n id

Kerry350 committed:
Merge branch '235508-nested-conditionals-followups' of github.com:Kerry350/kibana into 235508-nested-conditionals-followups

Kerry350 committed:
Merge remote-tracking branch 'upstream/main' into 235508-nested-conditionals-followups

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/generate codeowners'

Kerry350 committed:
Fix change from merge crossover

Kerry350 committed:
Merge branch '235508-nested-conditionals-followups' of github.com:Kerry350/kibana into 235508-nested-conditionals-followups

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343559)
* Commit: 131bfc74d0e046b13f4d1d70d3a6294bd5d4d1c4

### Failed CI Steps
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343559/jobs/01998029-a47f-45c8-a71a-e2a973b914d7/artifacts/01998043-d540-49b8-9931-a4577881e81d) Scout: [ platform / streams_app ] plugin / serverless-oblt - Stream data processing - creating steps - should be able to nest steps under conditions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343559/jobs/01998029-a47f-45c8-a71a-e2a973b914d7/artifacts/01998043-d53f-4656-8d3c-73a42aa3e384) Scout: [ platform / streams_app ] plugin / serverless-oblt - Stream data processing - creating steps - should create a new condition successfully
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343559/jobs/01998029-a47f-45c8-a71a-e2a973b914d7/artifacts/01998043-d53f-45c8-a9f1-e60bb90648e6) Scout: [ platform / streams_app ] plugin / serverless-oblt - Stream data processing - editing steps - should edit an existing condition
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343559/jobs/01998029-a47f-45c8-a71a-e2a973b914d7/artifacts/01998043-d53f-440a-b807-dfc7fa22f080) Scout: [ platform / streams_app ] plugin / serverless-oblt - Stream data processing - error handling and recovery - should handle network failures during a processor creation
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343559/jobs/01998029-a47f-45c8-a71a-e2a973b914d7/artifacts/01998043-d53e-4599-abf9-db597b87b536) Scout: [ platform / streams_app ] plugin / serverless-oblt - Stream data processing - error handling and recovery - should recover from API errors during a processor updates
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343559/jobs/01998029-a47f-45c8-a71a-e2a973b914d7/artifacts/01998043-d53e-4662-ac40-8a5a59aeb961) Scout: [ platform / streams_app ] plugin / stateful - Stream data processing - creating steps - should be able to nest steps under conditions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343559/jobs/01998029-a47f-45c8-a71a-e2a973b914d7/artifacts/01998043-d540-4b5d-bb98-8de6e2531789) Scout: [ platform / streams_app ] plugin / stateful - Stream data processing - creating steps - should create a new condition successfully
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343559/jobs/01998029-a47f-45c8-a71a-e2a973b914d7/artifacts/01998043-d53e-4c03-a463-d678ddc7208a) Scout: [ platform / streams_app ] plugin / stateful - Stream data processing - editing steps - should edit an existing condition
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343559#01998029-a47f-45c8-a71a-e2a973b914d7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343559/jobs/01998029-a47f-45c8-a71a-e2a973b914d7/artifacts/01998043-d540-4a02-b5cd-2a90825c4f71) Scout: [ platform / streams_app ] plugin / stateful - Stream data processing - error handling and recovery - should recover from API errors during a processor updates

### History
* :broken_heart: [Build #343293](https://buildkite.com/elastic/kibana-pull-request/builds/343293) failed 65369c1f1bb236a7b324c8d5f88f2b6264e46b1c
* :broken_heart: [Build #343173](https://buildkite.com/elastic/kibana-pull-request/builds/343173) failed 1978b53b2c03db878f4693e47b6efafee6b7c050

cc @Kerry350

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343559","number":343559,"commit":"131bfc74d0e046b13f4d1d70d3a6294bd5d4d1c4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343293","number":343293,"commit":"65369c1f1bb236a7b324c8d5f88f2b6264e46b1c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343173","number":343173,"commit":"1978b53b2c03db878f4693e47b6efafee6b7c050"}],"number":343559}
buildkite-pr-comment-->

Kerry350 committed:
Merge remote-tracking branch 'upstream/main' into 235508-nested-conditionals-followups

Kerry350 committed:
Amend tests to match new functionality

Some functionality changed in other PR merges, and Scout tests don't run on every PR build, this should align everything.

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343727)
* Commit: cc82fac5c3d80a25291c18afee38f4ee4cf10920

### Failed CI Steps
* [FTR Configs #86](https://buildkite.com/elastic/kibana-pull-request/builds/343727#01998105-0ddd-4510-bb16-766f323493f9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343727#01998105-0ddd-4510-bb16-766f323493f9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343727/jobs/01998105-0ddd-4510-bb16-766f323493f9/artifacts/0199813a-2390-43d7-bad7-53722c0b4a3f) FTR Configs #<span></span>86 / Core Analysis - Entity Store @<span></span>ess CRUD API - Upsert upsert user Should update an existing user

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/cdee51bd60f1e45f994ac887a181f6b726a783df) | [after](https://github.com/elastic/kibana/commit/cc82fac5c3d80a25291c18afee38f4ee4cf10920) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 832.6KB | 833.0KB | +481.0B |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/cdee51bd60f1e45f994ac887a181f6b726a783df) | [after](https://github.com/elastic/kibana/commit/cc82fac5c3d80a25291c18afee38f4ee4cf10920) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 12 | 13 | +1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/cdee51bd60f1e45f994ac887a181f6b726a783df) | [after](https://github.com/elastic/kibana/commit/cc82fac5c3d80a25291c18afee38f4ee4cf10920) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 23 | 24 | +1 |


</details>


### History
* :broken_heart: [Build #343559](https://buildkite.com/elastic/kibana-pull-request/builds/343559) failed 131bfc74d0e046b13f4d1d70d3a6294bd5d4d1c4
* :broken_heart: [Build #343293](https://buildkite.com/elastic/kibana-pull-request/builds/343293) failed 65369c1f1bb236a7b324c8d5f88f2b6264e46b1c
* :broken_heart: [Build #343173](https://buildkite.com/elastic/kibana-pull-request/builds/343173) failed 1978b53b2c03db878f4693e47b6efafee6b7c050

cc @Kerry350

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343727","number":343727,"commit":"cc82fac5c3d80a25291c18afee38f4ee4cf10920"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343559","number":343559,"commit":"131bfc74d0e046b13f4d1d70d3a6294bd5d4d1c4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343293","number":343293,"commit":"65369c1f1bb236a7b324c8d5f88f2b6264e46b1c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343173","number":343173,"commit":"1978b53b2c03db878f4693e47b6efafee6b7c050"}],"number":343727}
buildkite-pr-comment-->
